usecases/schema/tenant.go

```diff
@@ -292,3 +292,30 @@ func IsLocalActiveTenant(phys *sharding.Physical, localNode string) bool {
 	return slices.Contains(phys.BelongsToNodes, localNode) &&
 		phys.Status == models.TenantActivityStatusHOT
 }
+
+func (h *Handler) getTenantsByNames(class string, names []string) ([]*models.Tenant, error) {
+	info, err := h.multiTenancy(class)
+	if err != nil || info.Tenants == 0 {
+		return nil, err
+	}
+
+	ts := make([]*models.Tenant, 0, len(names))
+	f := func(_ *models.Class, ss *sharding.State) error {
+		if n := len(ss.Physical); n > len(ts) {
+			ts = make([]*models.Tenant, n)
+		} else if n < len(ts) {
+			ts = ts[:n]
+		}
```

Comment:
can you first deduce the size and then just create a `ts` slice with this appropriate size?
No you are creating a big silce and then if `n` is smaller then you are creating a new smaller slice. I think that you can first deduce the size and the only create the slice once with appropriate size

Reply:
Ah, this code is actually completely irrelevant to `getTenantsByNames`. It's a relic of me copying `getTenants` and not appropriately removing it.

`getTenantsByNames` will return list of length `<=len(names)` depending on whether the `name` was found in the local shard or not with the variable defined as `ts := make([]*models.Tenant, 0, len(names))` on `L302`. So this if statement will always be `true` and it's therefore irrelevant. I will remove it!